/* ----------------------------------------------------------------------------
 * telemetry.proto - Telemetry protobuf definitions
 *
 * August 2015, Robert Wills
 *
 * Copyright (c) 2015 by Cisco Systems, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ----------------------------------------------------------------------------
 */

message TelemetryHeader {
  optional uint32 encoding = 1;

  optional string policy_name = 2;
  optional string version = 3;
  optional string identifier = 4;

  optional uint64 start_time = 5;
  optional uint64 end_time = 6;
  
  repeated TelemetryTable tables = 7;
}

message TelemetryTable {
  optional string policy_path = 1;
  repeated bytes  row = 2;
}

